frontend/src/views/new-project/NewProjectView.tsx

```diff
@@ -60,66 +60,101 @@ const createProjectMutation = gql`
  */
 export const NewProjectView = () => {
   const [createProject, { loading, data, error }] = useMutation(createProjectMutation);
-  const [state, dispatch] = useReducer(reducer, initialState);
-
-  const { viewState, name, nameMessage, nameMessageSeverity, message, newProjectId } = state;
+  const [{ value, context }, dispatch] = useMachine<NewProjectViewContext, NewProjectEvent>(newProjectViewMachine);
+  const { newProjectView, toast } = value as SchemaValue;
+  const { name, newProjectId, message } = context;
 
   const onNameChange = (event) => {
-    const name = event.target.value;
-    dispatch({ type: HANDLE_CHANGE_NAME__ACTION, name });
+    const value = event.target.value;
+    const changeNameEvent: ChangeNameEvent = { type: 'CHANGE_NAME', name: value };
+    dispatch(changeNameEvent);
   };
 
   const onCreateNewProject = async (event) => {
     event.preventDefault();
-
     const variables = {
       input: {
         name: name.trim(),
         visibility: 'PUBLIC',
       },
     };
+    const submitEvent: SubmitEvent = { type: 'SUBMIT' };
+    dispatch(submitEvent);
     createProject({ variables });
   };
 
   useEffect(() => {
     if (!loading && !error && data) {
       if (error) {
-        dispatch({ type: HANDLE_SUBMIT_FAILURE__ACTION, reponse: error });
+        let errorMessage = error.message;
+        const showToastEvent: ShowToastEvent = { type: 'SHOW_TOAST', message: errorMessage };
+        dispatch(showToastEvent);
       } else if (data) {
-        dispatch({ type: HANDLE_SUBMIT__ACTION, response: data });
+        let id = data.createProject?.project?.id;
+        if (id) {
+          const HandleSuccessEvent: HandleSuccessEvent = { type: 'HANDLE_SUCCESS', id };
+          dispatch(HandleSuccessEvent);
+        }
+        else if (data.createProject?.message) {
+          const handlErrorEvent: HandleErrorEvent = { type: 'HANDLE_ERROR' };
+          dispatch(handlErrorEvent);
+          dispatch({ type: 'SHOW_TOAST', message: data.createProject?.message });
+        }
       }
     }
-  }, [loading, data, error]);
+  }, [loading, data, error, dispatch]);
 
-  if (newProjectId) {
+  if (newProjectView === 'success') {
     return <Redirect to={`/projects/${newProjectId}/edit`} />;
   }
 
-  const canBeSubmitted = viewState === VALID__STATE;
-  const nameMessageContent = nameMessage ? (
-    <Message severity={nameMessageSeverity} data-testid="nameMessage" content={nameMessage} />
-  ) : null;
+  const canBeSubmitted = newProjectView === 'valid';
+  let validationMessage = '';
+  if (newProjectView === 'invalid') {
+    validationMessage = 'The name must contain between 3 and 20 characters.';
+  }
```

Comment:
This logic should be in the state machine. After some discussion with Laurent, it would be better to have the following fields in your context : name, nameMessage, isNameError or something like that